### ISSUE TITLE
Introducing SIoU Loss

### ISSUE THREAD

**Issue by WZMIAOMIAO** [ultralytics/yolov5#8138]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar feature requests.

### Description

Hi, have you ever tried SIoU Loss? 

Paper: [SIoU Loss: More Powerful Learning for Bounding Box Regression](https://arxiv.org/abs/2205.12740)

According to the Abstract of the paper, feel so good.

> In particular, the application of SIoU to the COCO-train/COCO-val results in improvements of +2.4% (mAP@0.5:0.95) and +3.6%(mAP@0.5) over other Loss Functions.

![image](https://user-images.githubusercontent.com/31005897/172523112-2e102bf8-5959-4121-9df8-67d78b2792b5.png)

**Comment by wenjunx** [comment]
in my datasets, it didn't work.

**Comment by wilile26811249** [comment]
I just started to try SIOU Loss.
I'll report if the training have any news.

**Comment by wilile26811249** [comment]
Other discussion about the SIOU loss in the PaddleDetection.
https://github.com/PaddlePaddle/PaddleDetection/issues/6070

It doesn't seem as good as in the paper.

**Comment by glenn-jocher** [comment]
@wilile26811249 interesting, let us know your experiment results!

**Comment by wilile26811249** [comment]
Ok, Sure.
But, I am training on the modified YOLOv5s(custom model).
I had trained that with CIOU before, there can be a comparison.

Training takes about a week. (Device: 2*2080ti Epoch: 300)

**Comment by glenn-jocher** [comment]
@AyushExel can you take a look at this SIoU loss to see if it might be suitable for YOLOv5?

@wilile26811249 do you have a torch SIoU implementation we could use to experiment with?

The official publication lacks code unfortunately.

**Comment by wilile26811249** [comment]
You can check this.

https://github.com/xialuxi/yolov5-car-plate/blob/78cd5d46ca1f140a8bc154f936b0dffbf7e0b087/utils/general.py#L370-L388

**Comment by wilile26811249** [comment]
You can try this implement.

https://github.com/xialuxi/yolov5-car-plate/blob/78cd5d46ca1f140a8bc154f936b0dffbf7e0b087/utils/general.py#L370-L388

**Comment by wilile26811249** [comment]
You can check this implementation.

https://github.com/xialuxi/yolov5-car-plate/blob/master/utils/general.py#L370-L388

**Comment by AyushExel** [comment]
@glenn-jocher  I think the correct way forward for supporting this in the master branch would be to run ablation tests. For starters we can run a training job comparing SIoU with current version on 10% coco dataset and then on the full dataset. Ideally, if it outperforms on both tests then we'll have a clear indication( if it doesn't affect other things like speed) otherwise we can just add the implementation and allow user to enable it via a feature flag.
This could be a nice community led R&D experiment.

**Comment by wilile26811249** [comment]
@glenn-jocher @WZMIAOMIAO, Here are my results. 
Unfortunately, It seems like not improved.

with SIOU:
<img width="533" alt="image" src="https://user-images.githubusercontent.com/23655220/174104581-c58bd208-3374-4fe9-b16e-e355b5899395.png">

with CIOU:
<img width="705" alt="image" src="https://user-images.githubusercontent.com/23655220/174104395-1e2fb75a-a634-4164-b1c0-9e43d2e3aa10.png">

**Comment by glenn-jocher** [comment]
@wilile26811249 thanks for the results! Ok so it seems like not a significant change.

**Comment by WZMIAOMIAO** [comment]
well, I will close this issue.

**Comment by lucysumi** [comment]
> I just started to try SIOU Loss. I'll report if the training have any news.

Hi, @wilile26811249 Could you please tell me how do you change the current Yolov5 loss function fro CIOU to SIOU loss? Where specifically do I make the changes? I'm a newbie and trying to understand Yolov5. Thanks!
